extensions/origins/source/protocols/pubsub/Topic.ts

```diff
@@ -0,0 +1,23 @@
+import { Connector } from '@toa.io/core'
+import type { Origin } from './Origin'
+import type { PublishOptions, PubSub } from '@google-cloud/pubsub'
+
+export class Topic extends Connector {
+  private readonly publisher
+
+  public constructor (pubsub: PubSub, origin: Origin) {
+    super()
+
+    const name = `projects/${origin.project}/topics/${origin.topic}`
+
+    this.publisher = pubsub.topic(name, OPTIONS)
+  }
+
+  public async publish (payload: unknown): Promise<string> {
+    const data = Buffer.from(JSON.stringify(payload))
+
+    return this.publisher.publishMessage({ data })
```

Comment:
I believe simplification of JSON.stringify is not a generic frameworks applicable approach. 

Ideally, we should query the topic to understand format:
```js
 // Get the topic metadata to learn about its schema encoding.
  const topic = pubSubClient.topic(topicNameOrId);
  const [topicMetadata] = await topic.getMetadata();
  const topicSchemaMetadata = topicMetadata.schemaSettings;
```

Then there are documentation how to publish in [Avro](https://cloud.google.com/pubsub/docs/samples/pubsub-publish-avro-records) or [protobuf](https://cloud.google.com/pubsub/docs/samples/pubsub-publish-proto-messages)

Reply:
https://github.com/toa-io/toa/pull/604/commits/21b411aca23d60b3182eaa1928bf6fcd48eeec74

Relying on topic settings is obviously more generic approach. But it will force us to implement all formats supported by Pub/Sub (and keeping this up-to-date) which is simply unreasonable amount of work especially considering that no one will ever care.